ITR/data/base_providers.py

```diff
@@ -588,16 +586,17 @@ def _standardize(self, intensities: pd.DataFrame) -> pd.DataFrame:
         # When columns are years and rows are all different intensity types, we cannot winsorize
         # Transpose the dataframe, winsorize the columns (which are all coherent because they belong to a single variable/company), then transpose again
         intensities = intensities.T
-        for col in intensities.columns:
-            pa = PA_._from_sequence(intensities[col])
-            pa_units = pa[0].u
-            pa_nans = unp.isnan(pa.data)
-            if pa_nans.any():
-                if [isinstance(pt.m, UFloat) for pt in pa[np.where(pa_nans)]]:
-                    u_pa_data = [_ufloat_nan if pn else pt if isinstance(pt, UFloat) else ufloat(pt, 0)
-                                 for pt, pn in zip(pa.data, pa_nans)]
-                    intensities[col] = pd.Series(PA_(u_pa_data, pa_units), dtype=pa.dtype,
-                                                 index=intensities[col].index, name=intensities[col].name)
+        if ITR.HAS_UNCERTAINTIES:
+            for col in intensities.columns:
+                pa = PA_._from_sequence(intensities[col])
+                pa_units = pa[0].u
+                pa_nans = ITR.isnan(pa.data)
+                if pa_nans.any():
+                    if [isinstance(pt.m, UFloat) for pt in pa[np.where(pa_nans)]]:
+                        u_pa_data = [_ufloat_nan if pn else pt if isinstance(pt, UFloat) else ufloat(pt, 0)
+                                     for pt, pn in zip(pa.data, pa_nans)]
+                        intensities[col] = pd.Series(PA_(u_pa_data, pa_units), dtype=pa.dtype,
+                                                     index=intensities[col].index, name=intensities[col].name)
```

Comment:
looks like, this way, we completely skip `intensities`, if HAS_UNCERTAINTIES is False

Reply:
This code is merely doing by hand a kind of numerical promotion that only needs to be done if we find that our incoming series has some uncertainty values mixed with NaNs.  The code could be written so that ALL numbers, upon initial parsing, are promoted to uncertainties (+/- 0.0), but that adds a lot of overhead to cases where we don't need it.

Presently, the code uses NaN to represent NULL values (a hold-over from pre-1.0 versions of Pandas).  Lots of Pandas code does this, though in the modern era pd.NA (or pd.NAT for text) is a perfectly reasonable NULL value.  The Pint-Pandas folks are just about finishing changes so that the PintArray ExtensionArray will have a fully-working pd.NA value for its type, and when that happens, we don't need to create a Quantity(NaN, unit) but may be able to just have pd.NA as a generic NULL value.

In any case, this code is all about normalizing the NaNs, representing NULL, between the pd.nan world (pure float) and the UFloat world.  We don't need to do any work if there are no NaNs in the particular series.